internal/runners/revert/revert.go

```diff
@@ -72,28 +73,50 @@ func (r *Revert) Run(params *Params) error {
 	if !strfmt.IsUUID(params.CommitID) {
 		return locale.NewInputError("err_invalid_commit_id", "Invalid commit ID")
 	}
+	if params.CommitID == r.project.CommitID() && params.To {
+		return locale.NewInputError("err_revert_to_current_commit", "Cannot revert to current commit")
+	}
 	r.out.Notice(locale.Tl("operating_message", "", r.project.NamespaceString(), r.project.Dir()))
 	commitID := strfmt.UUID(params.CommitID)
 
-	priorCommits, err := model.CommitHistoryPaged(commitID, 0, 2)
-	if err != nil {
-		return locale.WrapError(err, "err_revert_get_commit", "Could not fetch commit details for commit with ID: {{.V0}}", params.CommitID)
-	}
-	if priorCommits.TotalCommits < 2 {
-		return locale.NewInputError("err_revert_no_history", "Cannot revert commit {{.V0}}: no prior history", params.CommitID)
+	var targetCommit *mono_models.Commit
```

Comment:
I think I need some help understanding what the `targetCommit` represents in this code. We have a `from` and `to` commit so what is the target commit? It seems to only be used for sourcing author information and printing, except in the `to` case where it becomes the `toCommit`...

Reply:
`targetCommit` is either the commit whose contents are being reverted, or the commit being reverted to. It depends on the context of course. Yes, it is very confusing :/